### PR TITLE
doc: remove hacking-state.md which relates of mongodb

### PR DESCRIPTION
**Pull request by gfouillet**
- Before this commit, the documentation was referring to MongoDB and `state` package which will be removed in juju 4, in favor of DQLite and `domain` package.
- After this commit, the file `hacking-state.md` has been removed because outdated, and `architectural-overview.md` have been updated to remove relation to this file, `state` package or MongoDB and replaced by insight about DQLite and `domain` package.

## Links

**Launchpad bug:** https://bugs.launchpad.net/juju/+bug/

**Jira card:** [JUJU-4996](https://warthogs.atlassian.net/browse/JUJU-4996)

[JUJU-4996]: https://warthogs.atlassian.net/browse/JUJU-4996?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by SimonRichardson — APPROVED** [review]

File: doc/architectural-overview.md [hunk 1]
```diff
@@ -108,33 +108,23 @@ different set of workers to a machine agent.
...
 There's a lot of *detail* to cover, but there's not much to say from an architectural
-standpoint. We use a mongodb replicaset to support HA; we use the `mgo` package from
-`labix.org` to implement multi-document transactions; we make use of the transaction
-log to detect changes to particular documents, and convert them into business-object-
-level events that get sent over the API to interested parties.
+standpoint. We use DQLite to store the domain state and support HA.
+
+The domain package consists of several component package, stored in two SQL schema,
+one for the controller level, another for the model level. Each component provides a
+state layer to deal with the database and a service layer to provide access to client.
```
Comment (tmihoc) [inline doc/architectural-overview.md:118]: several ... packages* 

two SQL schemas*

and one* for the model level

to the* client

**Review by tmihoc — APPROVED** [review]
This doc could definitely be improved further, but the changes you've made are all that's strictly necessary for now.

File: doc/architectural-overview.md [hunk 2]
```diff
@@ -108,33 +108,23 @@ different set of workers to a machine agent.
 There are a number of *clients* which interact with Juju using the Juju
 API. These include the `juju` command line tool and Juju Dashboard.
```
[the remark this answers is not in the document]
Comment (gfouillet) [reply doc/architectural-overview.md:109]: sure, but i try to be as focused as possible on this PR.

File: doc/architectural-overview.md (hunk 1 above)
Comment (gfouillet) [reply doc/architectural-overview.md]: Thanks, spellfix done !

**Pull request merged**